File: C/BinarySearch.c
#include <stdio.h>

int binarySearch(int arr[], int left, int right, int target) {
    while (left <= right) {
        int mid = left + (right - left) / 2;

        if (arr[mid] == target) {
            return mid;
        }

        if (arr[mid] < target) {
            left = mid + 1;
        }
        else {
            right = mid - 1;
        }
    }

    // Target not found in the array
    return -1;
}

int main() {
    int arr[] = {12, 23, 34, 45, 56, 78, 89};
    int n = sizeof(arr) / sizeof(arr[0]);
    int target = 56;

    int result = binarySearch(arr, 0, n - 1, target);

    if (result != -1) {
        printf("Element %d found at index %d.\n", target, result);
    } else {
        printf("Element %d not found in the array.\n", target);
    }

    return 0;
}

File: JavaScript/binarySearch.js
function binarySearch(arr, target) {
    let left = 0;
    let right = arr.length - 1;
  
    while (left <= right) {
      const mid = Math.floor((left + right) / 2);
  
      if (arr[mid] === target) {
        return mid; // Element found, return its index
      }
  
      if (arr[mid] < target) {
        left = mid + 1; // Adjust the left boundary
      } else {
        right = mid - 1; // Adjust the right boundary
      }
    }
  
    return -1; // Element not found
  }
  
  // Example usage:
  const sortedArray = [2, 4, 6, 8, 10, 12, 14, 16, 18];
  const target = 10;
  const result = binarySearch(sortedArray, target);
  
  if (result !== -1) {
    console.log(`Element ${target} found at index ${result}`);
  } else {
    console.log(`Element ${target} not found in the array`);
  }
  